server/pbench/bin/index-pbench

```diff
@@ -187,99 +319,303 @@ def es_index(es, actions, dbg=0):
+            # names) with the identifiers as additional metadata. Note that we
+            # are constructing this document just from the current row of data
+            # taken from all .csv files (assumes timestamps are the same).
+            for fname,row in rows.items():
+                klass, metric = metric_mapping[fname]
+                for idx,val in enumerate(row):
+                    if idx == 0:
+                        continue
+                    # Given an fname and a column offset, return the
+                    # identifier from the header
+                    identifier, subfield = field_mapping[fname][idx]
+                    if subfield:
+                        if metric not in datum[identifier][self.toolname][klass]:
+                            datum[identifier][self.toolname][klass][metric] = _dict_const()
+                        datum[identifier][self.toolname][klass][metric][subfield] = val
+                    else:
+                        datum[identifier][self.toolname][klass][metric] = val
+            # At this point we have fully mapped all data from all .csv files
+            # to their proper fields for each identifier. Now we can yield
+            # record for the identifiers.
+            for key,source in datum.items():
+                yield source
+        return
 
-def mk_metadata(fname, tb, mdconf, md5sum):
-    """Return a dict with metadata about a tarball"""
-    mtime = datetime.fromtimestamp(os.stat(fname)[stat.ST_MTIME])
+    def _make_source_individual(self):
+        """Read .csv files individually, emitting records for each row and
+        column coordinate."""
+        pass
 
-    mddict = {'generated-by': _NAME_,
-              'generated-by-version': _VERSION_,
-              'pbench-agent-version': mdconf.get("pbench", "rpm-version"),
-              'file-date': mtime.strftime("%Y-%m-%d"),
-              'file-name': fname,
-              'md5': md5sum
-    }
-    return mddict
+    def make_source(self):
+        """Simple jump method to pick the write source generator based on the
+        handler's prospectus."""
+        if not self.csv_files:
+            # If we do not have any csv files for this tool, ignore it.
+            return
+        if self.handler['@prospectus']['handling'] == 'unify':
+            gen = self._make_source_unified()
+        elif self.handler['@prospectus']['handling'] == 'individual':
+            gen = self._make_source_individual()
+        else:
+            raise Exception("Logic bomb!")
+        return gen
 
+    @staticmethod
+    def get_csv_files(handler, iteration, sample, host, tool, mdlog, tb):
+        """
+        Fetch the list of .csv files for this tool, fetch their headers, and
+        return a dictionary mapping their column headers to their field names. 
+        """
+        # FIXME: We need to be able to handle other tool groups
+        path = os.path.join(iteration, sample, "tools-default", host, tool, "csv")
```

Comment:
This is also a pretty important step to handle.  @chaitanyaenr, can you take a look at what it would take to do this on `master` or the `wip-indexing` branch ahead of this commit?